### PR TITLE
Quoted tweets

### LINKED ISSUE #115: original author (if quoted retweet)
As stated in the requirements:

Each message must provide:
- ~~Link to the original message~~
- ~~Link to the authors page~~
- ~~time date~~ and original author (if retweeted)

I think it makes sense to add quoted tweets in...

### LINKED ISSUE #131: Only fuzzy search if a word is more than 5 characters
A user is unlikely to search "ManU" wrong, but likely to search "Manchester" incorrectly

### LINKED ISSUE #133: Tweets are missing author's profile picture

### PR DESCRIPTION
**Pull request by Winwardo**
Closes #115 
Closes #133 
Closes #131 

@VasilyShelkov ready for comment and merge

### CODE REVIEW

File: src/shared/data/databaseInsertActions.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ export const upsertTweeter = (db, tweeter) => {
 export const upsertTweet = (db, tweet) => (
   runQueryOnImmutableObject(
     db,
-    'UPDATE tweet SET id=:id, content=:content, date=:date, likes=:likes, retweets=:retweets, longitude=:longitude, latitude=:latitude UPSERT WHERE id=:id',
+    'UPDATE tweet SET id=:id, content=:content, date=:date, likes=:likes, retweets=:retweets, longitude=:longitude, latitude=:latitude, contains_a_quoted_tweet=:contains_a_quoted_tweet UPSERT WHERE id=:id',
```
Comment (Winwardo) [inline src/shared/data/databaseInsertActions.js]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2364107871/issues/source?bid=3148782&fileBranchId=3220697#l37)

Comment (Winwardo) [inline src/shared/data/databaseInsertActions.js]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2364358747/issues/source?bid=3148782&fileBranchId=3220697#l37)

File: src/client/results/Feed.js [hunk 2]
```diff
@@ -219,4 +228,56 @@ const Tweet = ({ content }) => {
   );
 };
```
Comment (VasilyShelkov) [inline src/client/results/Feed.js:230]: change to normal class

File: src/server/server.js [hunk 3]
```diff
@@ -39,6 +39,10 @@ app.get('/orient/generate', (req, res) => {
   generateDatabase(res);
 });
 
+app.get('/tweet/quotedby/:id', (req, res) => {
+  getQuotedTweetFromParent(req, res, req.params.id);
```
Comment (VasilyShelkov) [inline src/server/server.js]: any reason you're passing in the req and the params id then using req.params.id in getQuotedTweetFromParent

File: src/server/tweetFinder.js [hunk 4]
```diff
@@ -205,3 +228,28 @@ const getTweetsAsResults = (data) => (
...
 );
+
+export const getQuotedTweetFromParent = (req, res, id) => (
+  newPromiseChain()
+    .then(() => (
+       db.query(
+         makeTweetQuerySelectingFrom('TRAVERSE OUT FROM (SELECT OUT(\'QUOTED\') FROM (SELECT FROM Tweet WHERE id = :id))'),
+         {
+           params: {
+             id: req.params.id,
```
Comment (VasilyShelkov) [inline src/server/tweetFinder.js]: any reason you're passing in the req and the params id then using req.params.id in getQuotedTweetFromParent

File: src/server/server.js (hunk 3 above)
Comment (Winwardo) [inline src/server/server.js]: Was to be explicit, but I can take it out

File: src/shared/data/databaseInsertActions.js (hunk 1 above)
Comment (Winwardo) [inline src/shared/data/databaseInsertActions.js]: Codacy Issue found: [Line is too long.](https://www.codacy.com/app/winwardo/solid-octo-disco/file/2364574296/issues/source?bid=3148782&fileBranchId=3220697#l37)

**Pull request merged**